### PR TITLE
tests: Add integration test for HAProxy

### PR DESCRIPTION
**Pull request by luise**
This commit adds an integration test for the updated HAProxy blueprint. It should not be merged until the [updated HAProxy blueprint](https://github.com/quilt/haproxy/pull/6) has been merged.

### CODE REVIEW

**Comment by kklin** [comment]
Assigning this to @ejj as the same person should probably review the blueprint change and the test.

I'd be happy to review both if you're busy, @ejj.

File: quilt-tester/tests/haproxy/check_haproxy.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+		"a2": {},
+	}
+
+	expectedB := map[string]struct{}{
+		"b1": {},
+		"b2": {},
+		"b3": {},
+	}
+
+	getTestA := httpGetTest("serviceA.com", HAProxyIPs, expectedA)
```
Comment (ejj) [inline quilt-tester/tests/haproxy/check_haproxy.go]: I think it'd be worth doing this test in a loop a couple of times just to make sure the results are consistent across multiple requests.

Comment (ejj) [inline quilt-tester/tests/haproxy/check_haproxy.go]: Ah nvm, you do that internally.

File: quilt-tester/tests/haproxy/haproxy.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const quilt = require("@quilt/quilt");
```
Comment (ejj) [inline quilt-tester/tests/haproxy/haproxy.js]: Please lint this.

File: quilt-tester/tests/haproxy/haproxy.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const hap = require("@quilt/haproxy");
+var infrastructure = require("../../config/infrastructure.js");
+
+const indexPath = "/usr/share/nginx/html/index.html";
+
+function containerWithContent(content) {
+	var files = {};
```
Comment (ejj) [inline quilt-tester/tests/haproxy/haproxy.js]: Should this be 2, 4, or 8 spaces?

**Comment by kklin** [comment]
Ah @luise, the test will need to be modified as in https://github.com/quilt/quilt/pull/961/commits/9052f0407f2d697970366e36c25e353a203d0953

**Comment by luise** [comment]
Updated. The README test in the build is expected to fail until the new HAProxy blueprint is merged.

**Pull request merged**